### PR TITLE
support for adding iptables

### PR DESCRIPTION
**Pull request by roymarantz**
This adds support for inserting content into your genesis boot image that is there at boot time.  One example of use is setting up iptables.

This code currently builds and image, but I haven't tested that the result "does the right thing".  As such I'm interest in comments on the approach and implementation mostly.

@byxorna @defect @gtorre @qx-xp @Mitsuwa comments?

### CODE REVIEW

File: bootcd/create-image.sh [hunk 1]
```diff
@@ -59,6 +59,23 @@ ns=`grep nameserver /etc/resolv.conf`
 nameserver 8.8.4.4'
 perl -pe "s/%%LocalNameservers%%/$ns/" genesis.ks.template > "$tmpdir/genesis.ks"
 
+# insert Extra Post Content into the %POST section of the ks file
+# used to build the image from file paths like /conf/*.content
+# which is bind mounted via: docker run -v /Some/Dir:/conf:ro ....
+if [[ $(find /conf -name \*.content 2>/dev/null | wc -l) > 0 ]]; then
+   echo '### inserting Extra Post Content'
+   sed -i -e "
```
Comment (qx-xp) [inline bootcd/create-image.sh]: I tried running this `sed` blob but I don't think it produce the intended result. More testing is needed.

File: README.md [hunk 2]
```diff
@@ -90,6 +90,13 @@ To make it easy to get started with genesis, we have included a `Dockerfile` tha
 # ls output
 ```
 
+If you need to add stuff to the image, i.e. before genesis starts, then create file(s), named `SOMETHING.content` in some directory and bind mount that directory as `/conf` when running genesis-builder.  This is a handy way to setup iptables in your image.  See bootcd/iptable.content.sample for an example.
```
[the remark this answers is not in the document]
Comment (roymarantz) [reply README.md]: suggestions welcomed

**Comment by roymarantz** [comment]
@qx-xp I've addressed your comments and tested the insertion code by extracting and running it on my laptop to verify that it works.

I'll do some complete image build as a final test and then declare this RFC.

**Comment by roymarantz** [comment]
I have verified that the code works with iptables rules from another repo.

bump!

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (qx-xp) [reply README.md]: let me try... 
"It is possible to customize the image with extra content specific to your site, such as adding iptables rules.
To do so, create a directory and put your custom code into one or more files named `<something>.content`.
Bind mount this directory as `/conf` when running genesis-builder.
Any file with the `.content` extension under `/conf` will be added to the build.
See bootcd/iptable.content.sample for an example."

**Comment by roymarantz** [comment]
I removed `set -e` from run-genesis-bootloader since `rvm.sh` was causing that script to fail for no easily desirable reason and the script does very little else that might benefit from it.

**Comment by qx-xp** [comment]
I agree that `set -e` does not work with `source /etc/profile.d/rvm.sh`. I support removing `set -e` from run-genesis-bootloader.

**Pull request merged**